frontends/infinite-corridor/src/pages/SearchPage.tsx

```diff
@@ -1,11 +1,152 @@
+const ALLOWED_TYPES = ["program", "course"]
+//@ts-expect-error SETTINGS exists
+const pageSize = SETTINGS.search_page_size
+const SEARCH_RESUTLT_CELL_WIDTH = 12
+
+interface Result {
+  _source: LearningResourceResult
+}
 
 const SearchPage: React.FC = () => {
-  const [searchParams] = useSearchParams()
+  const [results, setSearchResults] = useState<Result[]>([])
+  const [total, setTotal] = useState(0)
+  const [completedInitialLoad, setCompletedInitialLoad] = useState(false)
+  const [requestInFlight, setRequestInFlight] = useState(false)
+  const [searchApiFailed, setSearchApiFailed] = useState(false)
+  const [aggregations, setAggregations] = useState<Aggregations>(new Map())
+
+  const clearSearch = useCallback(() => {
+    setSearchResults([])
+    setCompletedInitialLoad(false)
+    setTotal(0)
+  }, [setSearchResults, setCompletedInitialLoad, setTotal])
+
+  const runSearch = useCallback(
+    async (text: string, activeFacets: Facets, from: number) => {
+      if (activeFacets["type"]) {
+        activeFacets["type"] = intersection(activeFacets["type"], ALLOWED_TYPES)
+      } else {
+        activeFacets["type"] = ALLOWED_TYPES
+      }
+      setRequestInFlight(true)
+      const newResults = await search({
+        text,
+        from,
+        activeFacets,
+        size: pageSize
+      })
+      setRequestInFlight(false)
+
+      if (!newResults || newResults["apiFailed"]) {
+        setSearchApiFailed(true)
+        return
+      }
+
+      setAggregations(new Map(Object.entries(newResults.aggregations ?? {})))
+
+      setSearchResults(results =>
+        from === 0 ?
+          newResults.hits.hits :
+          [...results, ...newResults.hits.hits]
+      )
+      setTotal(newResults.hits.total)
+      setCompletedInitialLoad(true)
+    },
+    [
+      setSearchResults,
+      setTotal,
+      setCompletedInitialLoad,
+      setRequestInFlight,
+      setSearchApiFailed
+    ]
+  )
+
+  const search = async (params: SearchQueryParams) => {
+    const body = buildSearchQuery(params)
+
+    return fetchJSONWithCSRF("/api/v0/search/", {
+      method: "post",
+      body:   JSON.stringify(body)
+    })
+  }
+
+  const { updateText, loadMore, text, onSubmit, from, clearText } =
+    useCourseSearch(
+      runSearch,
+      clearSearch,
+      aggregations,
+      // this is the 'loaded' value, which is what useCourseSearch uses
+      // to determine whether to fire off a request or not.
+      completedInitialLoad && !requestInFlight,
+      pageSize
+    )
+
   return (
-    <div className="page-content">
-      Search params: <code>{searchParams.toString()}</code>
+    <div>
+      <BannerPageWrapper>
+        <BannerPageHeader tall compactOnMobile>
+          <Searchbox
+            className="course-searchbox"
+            onChange={updateText}
+            value={text || ""}
+            onClear={clearText}
+            //@ts-expect-error - I'm not sure how to fix this
```

Comment:
I was having a lot of trouble getting the types for the onSubmit in SearchInput.tsx to match the event type from course-search-util. I would appreciate help fixing this error

Reply:
The types in `course-search-util` are wrong. I've been meaning to make a PR.

`course-search-util` *says* that `onSubmit` should be a function `(e: React.FormEvent<Element>) => void`, but in reality onSubmit is triggered from a few different ways, and none of them are FormEvents. (They're click events; they previously were Keyboard events and click events).

Within course-search-utils itself, the only things that are used on the submit event are `event.target.value` and `event.preventDefault`, and the preventDefault is called unnecessarily.